tests/dvbuilder/dv_builder.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package dvbuilder
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+)
+
+const (
+	dvRandomNameLength       = 12
+	PVCSizeForRegistryImport = "512Mi"
+)
+
+// DVOption is an option type for the DVBuilder
+type DVOption func(*v1beta1.DataVolume)
+
+// NewDataVolume Set up a new DataVolume with a random name, a namespace and an optional list of options
+func NewDataVolume(namespace string, options ...DVOption) *v1beta1.DataVolume {
+	name := "test-datavolume-" + rand.String(dvRandomNameLength)
+	dv := &v1beta1.DataVolume{
+		TypeMeta: v12.TypeMeta{
+			APIVersion: "cdi.kubevirt.io/v1beta1",
+			Kind:       "DataVolume",
+		},
+		ObjectMeta: v12.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+	}
+
+	for _, option := range options {
+		option(dv)
+	}
+
+	return dv
+}
+
+// WithPVC is a DVOption to add a PVCOption spec to the DataVolume
+func WithPVC(storageClass string, size string, accessMode v1.PersistentVolumeAccessMode, volumeMode v1.PersistentVolumeMode) DVOption {
```

Comment:
I think this could be better. Something like:
`func TargetPVC(options PVC options)` 
You can then default `storageClass`, `volumeMode`, `accessMode`,... 
I think there exist sensible defaults. @brybacki WDYT?

Reply:
`volumeMode` already has a k8s default, it is FileSystem, so we only need to specify volumeMode for BlockStorage
`storageClass` if not set, will be a defaultStorageClass (at least in our kubernetes) -> https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1.

We might look into `tests/libstorage/pvc.go` and `tests/libstorage/datavolume.go` and check what kind of PVC helpers exist there and which are used currently.   

I think this `WithPVC` is a good raw factory/option function. If we see a repeating pattern in the use, it will be easy to create more helpers.